packages/ckeditor5-package-generator/lib/utils/copy-files.js

```diff
@@ -56,73 +46,35 @@ module.exports = function copyFiles( logger, options ) {
 		logger.verboseInfo( `* Copying "${ chalk.gray( templatePath ) }"...` );
 
 		const data = {
-			cliSeparator: options.program === 'npm' ? '-- ' : '',
+			cliSeparator: options.packageManager === 'npm' ? '-- ' : '',
 			now: new Date(),
 			...options
 		};
 
 		copyTemplate( templatePath, options.directoryPath, data );
 	}
-
-	createGitignore( options );
 };
 
 /**
  * Copies all files into the package directory. If any file has any template placeholders, they are filled.
  *
  * @param {String} templateFile The relative path to the "templates/" directory of the file to copy.
  * @param {String} packagePath The destination directory where the new package is created.
- * @param {Object} [data] The data to fill in the template file.
+ * @param {Object} data The data to fill in the template file.
  */
 function copyTemplate( templateFile, packagePath, data ) {
 	const rawFile = fs.readFileSync( path.join( TEMPLATE_PATH, templateFile ), 'utf-8' );
 	const filledFile = template( rawFile )( data );
 
-	const destinationPath = path.join(
-		packagePath,
+	const destinationPath = path.join( packagePath, templateFile )
 		// Remove sub-directory inside templates to merge results into one directory.
-		templateFile.replace( /^(?:common|js|ts)(?:\\|\/)/, '' )
-	);
+		.replace( /(?:common|js|ts)(?:\\|\/)/, '' )
+		// We use the ".txt" file extension to circumvent syntax errors in templates and npm not publishing the ".gitignore" file.
+		.replace( /\.txt$/, '' )
+		// Replace placeholder filenames with the class name.
+		.replace( /_PLACEHOLDER_/, data.formattedNames.plugin.lowerCaseMerged );
```

Comment:
Please, add a test for the `copyFiles()` function that uses the `common` name as a user name. Also, add a test if a user executes the package generator inside the `Projects/` directory. Please, notice the `ts` letters.

Examples:

* `/Users/common/workspace/...` (rare case)
* `/Users/pomek/Projects/...` (my environment)

Both should pass.

The `replace()` statements use too broad a scope. The tool should adjust template paths and only them. 

I would stop using regexp everywhere. 

We can use the simplest solution to remove the first directory from the template path. Then, merge the destination path with the destination template path. No regexp, no problems.

```
const destinationTemplatePath = templateFile.split( path.sep ).slice( 1 ).join( path.sep );
```